lib/buildkite.sh

```diff
@@ -383,3 +385,48 @@ EOF
 EOF
   fi
 }
+
+##
+## Generate an annotation for the build based on the plan result
+##
+## This function looks for a plan file at "${build_dir}/terraform.tfplan"
+## If there is no file, we assume the plan failed, and we create an error annotation with a link to the failed job
+## If there is a file, we inspect the .resource_changes[].change.actions fields for each resource.
+## If all of these fields are "no-op", the plan has succeeded with no changes
+## If any of these fields are not "no-op", the plan has succeeded with changes. In this case, we do a terraform show
+## and render these changes in the annotation.
+##
+bk_plan_annotate() {
+  # Ensure that a workspace argument (-w/--workspace) was specified.
+  if [[ -z "${_arg_workspace:-}" ]]; then
+    die "No Terraform workspace specified. This command requires a --workspace argument."
+  fi
+
+  info_msg "Annotating build with plan output"
+
+  _tf_plan_file="${build_dir}/terraform.tfplan"
+  if [[ -f "${_tf_plan_file}" ]]; then
+    is_all_no_ops=$(terraform show -json "${_tf_plan_file}" | jq '[.resource_changes[].change.actions] | flatten | all(. == "no-op")')
```

Comment:
```suggestion
	local is_all_no_ops
    is_all_no_ops="$(terraform show -json "${_tf_plan_file}" | jq '[.resource_changes[].change.actions] | flatten | all(. == "no-op")')"
```

The thing with local variables is that you can't do `local var=$(cmd)` because if `cmd` fails the failure gets smothered and `set -e` has no effect. It's just a Bash oddity.